apps/src/templates/ShareWarningsDialog.jsx

```diff
@@ -0,0 +1,79 @@
+var ShareWarnings = require('./ShareWarnings.jsx');
+
+/**
+ * Modal for our SharingWarnings.
+ */
+var SharingWarningsDialog = module.exports = React.createClass({
+  propTypes: {
+    is13Plus: React.PropTypes.bool.isRequired,
+    showStoreDataAlert: React.PropTypes.bool.isRequired,
+    handleClose: React.PropTypes.func.isRequired,
+    handleTooYoung: React.PropTypes.func.isRequired
+  },
+
+  getInitialState: function() {
+    return { modalIsOpen: !this.props.is13Plus || this.props.showStoreDataAlert };
+  },
+
+  componentDidMount: function () {
+    // We didn't need to show our modal. Go through the close process so that
+    // app becomes unblocked
+    if (!this.state.modalIsOpen) {
+      this.handleClose();
+    }
+  },
+
+  handleClose: function() {
+    this.setState({modalIsOpen: false});
+    React.unmountComponentAtNode(this.getDOMNode().parentNode);
+    this.props.handleClose();
+  },
+
+  render: function () {
+    if (!this.state.modalIsOpen) {
+      return <div/>;
```

Comment:
Also call React.unmountComponentAtNode here? I'm not sure why you call it above, but it seems like we should be consistent between these two cases:
- dialog is rendered with `modalIsOpen: false`
- dialog is opened with `modalIsOpen: true` and then `handleClose` is called